src/main/java/smokefree/graphql/GraphqlFactory.java

```diff
@@ -51,9 +52,18 @@
 
     @Bean
     public ErrorExtensionsMapper<ValidationException> mapValidationExceptionToErrorExtension() {
-        return exceptionToErrorExtensionsMapper(ValidationException.class, e -> new ErrorExtensions(ErrorCode.VALIDATION, "Invalid data"));
+        return exceptionToErrorExtensionsMapper(ValidationException.class, e -> new ErrorExtensions(ErrorCode.VALIDATION, e.getMessage()));
+    }
+
+    @Bean
+    public ErrorExtensionsMapper<ExecutionException> mapDomainExceptionToErrorExtensions() {
+        return exceptionToErrorExtensionsMapper(ExecutionException.class, e -> new ErrorExtensions(ErrorCode.VALIDATION, e.getMessage().split(":")[1]));
```

Comment:
Let's discuss. I am curious where these exceptions come from and also whether we do not expose too much information if we send their message to the frontend